sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/samplers/SamplingResult.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.trace.samplers;
+
+import static java.util.Objects.requireNonNull;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.ReadableAttributes;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.api.trace.TraceState;
+import io.opentelemetry.context.Context;
+import java.util.List;
+
+/**
+ * Sampling result returned by {@link Sampler#shouldSample(Context, String, String, Span.Kind,
+ * ReadableAttributes, List)}.
+ */
+public interface SamplingResult {
+
+  /**
+   * Returns a {@link SamplingResult} with no attributes and {@link SamplingResult#getDecision()}
+   * returning {@code decision}.
+   *
+   * <p>This is meant for use by custom {@link Sampler} implementations.
+   *
+   * <p>Use {@link #create(Decision, Attributes)} if you need attributes.
+   *
+   * @param decision The decision made on the span.
+   * @return A {@link SamplingResult} with empty attributes and the provided {@code decision}.
+   */
+  static SamplingResult create(Decision decision) {
```

Comment:
Factories for `SamplingResult` moved to the `SamplingResult`

Reply:
Also I think using a method overload (same name) seems consistent with our normal style instead of the previous name `empty`